lib/Plack/Request.pm

```diff
@@ -132,6 +132,15 @@ sub headers {
     $self->{headers};
 }
 
+sub header_parameters {
+    my $self = shift;
+
+    return Hash::MultiValue->from_mixed(
+        map { $_ => +[ split /,\ */ => $self->headers->header($_) ] }
```

Comment:
That was because there was only one element.
I'll try to improve it and add the tests following the suggestions.